### PR TITLE
Caching state in auth service.

### PR DESCRIPTION
**Pull request by sebeder**
[Trello Card](https://trello.com/c/bF2oPtd7/698-langsamer-editor-lag-beim-editieren-von-modellen)

The problem was a databinding agains the cookie service that was called very often. However, it is expensive to do so. I cached the authentication state and that increased the editor performance.

### CODE REVIEW

**Comment by sebeder** [comment]
@fischJan , that's the PR I was talking about. I added, as discussed, also the different loading behavior.

File: web/src/app/modules/data/modules/data-service/services/data-cache.ts [hunk 1]
```diff
@@ -50,7 +50,7 @@ export class DataCache {
         return this.elementStore[url];
     }
 
-    public readContents(url: string): IContainer[] {
+    public readContents(url: string, omit?: IContainer[]): IContainer[] {
```
Comment (junkerm) [inline web/src/app/modules/data/modules/data-service/services/data-cache.ts]: What is the omit parameter for? It is never used

**Review by junkerm — COMMENTED** [review]
It seems to work, however the performance in a project with 50.000 items is still bad. It takes a long time to load the project explorer, to load a requirement and some operations (e.g. save a model) also take long and even failed sometimes.

To reproduce the saving error: 
1) Create a new model
2) Add two nodes and connet
3) rename one node (inplace in the editor)
4) save: 
5) Result: Takes forever (I got logged out at one point)

I think that the problem cames because initially all items are loaded by the tree (as there are no intermediate folders)

The editor itself and the testspecification editor however is perceivably faster!

File: web/src/app/modules/data/modules/data-service/services/data-cache.ts (hunk 1 above)
Comment (sebeder) [reply web/src/app/modules/data/modules/data-service/services/data-cache.ts]: It was unused and unnecessary. I removed it.

**Review by junkerm — APPROVED** [review]
this works - please merge

**Pull request merged**